debian/build_virtualenv

```diff
@@ -112,7 +109,7 @@ esac
 
 # build the log config file
 "${TARGET_PYTHON}" "${VIRTUALENV_DIR}/bin/generate_log_config" \
-        --output-file="${PACKAGE_BUILD_DIR}/etc/matrix-synapse/log.yaml"
+        --output-file="${PACKAGE_BUILD_DIR}/etc/matrix-synapse/log.config"
```

Comment:
If you're renaming a conffile in debian packaging, you need to add stuff to the pre/postinst to help users.

See https://manpages.debian.org/unstable/dpkg/dpkg-maintscript-helper.1.en.html for more.

Reply:
Also, if it's a yaml file, I think it should be called `*.yaml`. Your concern that synapse might treat it as a regular config file if you put it in the config directory is noted... so don't do that?

Reply:
> If you're renaming a conffile in debian packaging, you need to add stuff to the pre/postinst to help users.

Will look into implementing a helper for that.

> Also, if it's a yaml file, I think it should be called `*.yaml`. Your concern that synapse might treat it as a regular config file if you put it in the config directory is noted... so don't do that?

I think that's a foot gun I'd rather not implement. I'm sure that's an edge case, but since `--config-path=` can take directories and the config directory defaults to the directory of the last found config file I could foresee situations where (missing) config files are generated in the same directory where `--config-path=` is pointed to.